### PR TITLE
core, timePicker: fix the ability to control the component from parent

### LINKED ISSUE #80: [UI-CORE] make TimePicker controllable by parent component

### PR DESCRIPTION
**Pull request by kmer2016**
fix #80

### CODE REVIEW

File: ui-core/src/components/inputs/TimePicker.tsx [hunk 1]
```diff
@@ -1,48 +1,67 @@
...
+  className: string;
+  onSelectItem: (item: number) => void;
+};
 
-  const handleHourClick = useCallback((hour: string) => {
-    setSelectedHour(hour);
-  }, []);
+const TimeRange: React.FC<TimeRangeProps> = ({ range, selectedItem, className, onSelectItem }) => {
+  return (
+    <div className="time-grid">
```
Comment (SharglutDev) [inline ui-core/src/components/inputs/TimePicker.tsx:21]: Looks like the gap here is bigger than the one from sketch (should be 2px)

**Review by SharglutDev — CHANGES_REQUESTED** [review]

**Review by Yohh — COMMENTED** [review]
it seems we have a problem with `hours` & `minutes` props

they don't change anything without a refesh and they aren't absolute (set minus and reload, the timer collapse)

[Screencast from 2024-06-27 10-51-44.webm](https://github.com/OpenRailAssociation/osrd-ui/assets/20640252/81158a45-3a3e-473d-8a9f-73a2ad08cd09)

File: ui-core/src/components/inputs/TimePicker.tsx (hunk 1 above)
Comment (Yohh) [reply ui-core/src/components/inputs/TimePicker.tsx:21]: the value is weird, maybe an update of the mockup

[Screencast from 2024-06-27 10-58-03.webm](https://github.com/OpenRailAssociation/osrd-ui/assets/20640252/be59b4ee-313f-4e75-8e73-9a874fc18dc5)

File: ui-core/src/components/inputs/TimePicker.tsx [hunk 2]
```diff
@@ -1,48 +1,67 @@
...
 
-  const formattedHour = selectedHour.padStart(2, '0');
-  const formattedMinute = selectedMinute.padStart(2, '0');
-  const selectedTime = `${formattedHour}:${formattedMinute}`;
+  const formatTimeValue = (value: number, max: number) =>
+    Math.max(0, Math.min(max, value)).toString().padStart(2, '0');
 
-  const hours = [...Array(24).keys()].map((hour) => hour.toString().padStart(2, '0'));
-  const minutes = [...Array(12).keys()].map((minute) => (minute * 5).toString().padStart(2, '0'));
+  const selectedTime = `${formatTimeValue(hours, 23)}:${formatTimeValue(hours, 59)}`;
```
Comment (SharglutDev) [inline ui-core/src/components/inputs/TimePicker.tsx]:
```suggestion
  const selectedTime = `${formatTimeValue(hours, 23)}:${formatTimeValue(minutes, 59)}`;
```

**Review by SharglutDev — CHANGES_REQUESTED** [review]

File: ui-core/src/components/inputs/TimePicker.tsx (hunk 2 above)
Comment (kmer2016) [reply ui-core/src/components/inputs/TimePicker.tsx]: Done !

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested, good job !

**Pull request merged**